### PR TITLE
refactor: remove djs components and use /builders components instead

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Components from builders are reexported for convenience. 

Prerequisite:
- #7195 

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Comment by iCrawl** [comment]
Still having linting errors here.

File: packages/discord.js/src/structures/MessagePayload.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
+const { createComponent } = require('@discordjs/builders');
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/structures/MessagePayload.js:3]: due to this import position

**Comment by kyranet** [comment]
Please rebase, you have merge conflicts.

**Comment by iCrawl** [comment]
This needs a rebase.

File: packages/discord.js/src/structures/Message.js [hunk 2]
```diff
@@ -138,9 +138,9 @@ class Message extends Base {
     if ('components' in data) {
       /**
        * A list of MessageActionRows in the message
-       * @type {MessageActionRow[]}
+       * @type {ActionRow[]}
        */
-      this.components = data.components.map(c => BaseMessageComponent.create(c, this.client));
+      this.components = data.components.map(createComponent);
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/Message.js:143]: From what I heard this is much more performant than passing the method directly

```suggestion
      this.components = data.components.map(c => createComponent(c));
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Comment by suneettipirneni** [comment]
Don't merge yet, there's a performance pitfall that I need to address that occurs in the `MessagePayload` serialization. I'll fix this soon.

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**